packages/@aws-cdk/aws-ec2/lib/vpc-ref.ts

```diff
@@ -71,6 +72,11 @@ export interface IVpcNetwork extends IConstruct {
    */
   subnetIds(selection?: SubnetSelection): string[];
 
+  /**
+   * Return the subnets appropriate for the placement strategy
```

Comment:
Can we deprecate subnetIds? seems like those can be mapped from the subnet objects?

Reply:
If we still need it let’s rename to selectSubnetIds but I would just deprecate.

Reply:
You mean adding `@deprecated` in JSDoc and removing all calls to this method?

Reply:
Yap

Reply:
Nooo don't!

Reply:
I've put effort into precisely into deprecating access to subnet objects and just getting IDs instead, because that is basically all the class consumers ever need from this class anyway. The previous interface was wider than anyone needed, and uncomfortable for imported VPCs.

Reply:
Can you elaborate why returning IDs is a better API? It is in contradiction with one of our design tents to use strong types instead of IDs.

Reply:
First of all: convenience. A subnet is never what a consumer needs. It always needs a list of subnet IDs, to pass to CloudFormation. Why would you make a consumer grub around in the innards of my data structures, to do exactly the same operation in all places where it's called? I might as well do that for you.

Second of all: it allows us more flexilibity around exports and imports. The function:

```ts
function subnetIds(): string[];
```

Can be satisfied by a `{ Fn::ImportValue: Export }` or a `{ Ref: Parameter }`. The function:

```ts
function subnets(): ISubnet[]
```

Can not. Because of this, it's super convoluted to export/import VPCs across stacks today. It could all be dramatically simplified if we hadn't painted ourselves into a corner with an API that promises too much, for no good reason because all consumers do is map across those subnets to get the IDs anyway!

And if we're invoking design principles: it violates the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). The VPC is your abstraction, and you shouldn't depend on how it's implemented.